src/Microsoft.Azure.SignalR.Protocols/CloseWithAckMessage.cs

```diff
@@ -0,0 +1,117 @@
+        /// Gets or sets the list of excluded connection Ids.
+        /// </summary>
+        public IReadOnlyList<string> ExcludedList { get; set; }
+
+        public CloseMultiConnectionsWithAckMessage(int ackId) : base(ackId) { }
+    }
+
+
+    /// <summary>
+    /// A close-connection message.
+    /// </summary>
+    public class CloseConnectionWithAckMessage : CloseWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the connection Id for the connection.
+        /// </summary>
+        public string ConnectionId { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloseConnectionWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="connectionId">The connection Id.</param>
+        /// <param name="ackId">The ack Id for the message.</param>
+        public CloseConnectionWithAckMessage(string connectionId, int ackId) : base(ackId)
+        {
+            ConnectionId = connectionId;
+        }
+    }
+
+    /// <summary>
+    /// Close all the connections in the hub.
+    /// </summary>
+    public class CloseConnectionsWithAckMessage : CloseMultiConnectionsWithAckMessage
+    {
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloseConnectionsWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="ackId">The ack Id for the message.</param>
+        public CloseConnectionsWithAckMessage(int ackId) : base(ackId) { }
+    }
+
+    /// <summary>
+    /// Close connections for a user.
+    /// </summary>
+    public class CloseUserConnectionsWithAckMessage : CloseMultiConnectionsWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the user Id.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloseUserConnectionsWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="userId">The user Id for the message.</param>
+        /// <param name="ackId">The ack Id for the message.</param>
+        public CloseUserConnectionsWithAckMessage(string userId, int ackId) : base(ackId)
+        {
+            UserId = userId;
+        }
+    }
+
+    /// <summary>
+    /// Close connections in a group.
+    /// </summary>
+    public class CloseGroupConnectionsWithAckMessage : CloseMultiConnectionsWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the group name.
+        /// </summary>
+        public string GroupName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloseGroupConnectionsWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="groupName">The group name for the message.</param>
+        /// <param name="ackId">The ack Id for the message.</param>
+        public CloseGroupConnectionsWithAckMessage(string groupName, int ackId) : base(ackId)
+        {
+            GroupName = groupName;
```

Comment:
null check

Reply:
looks like there is no *null check* convention in service protocol